### PR TITLE
fix(internal): skip all files when listing store

### LINKED ISSUE #2297: Files are read as Models
**Describe the bug**

When a .DS_Store file is created in the `bentoml/models` folder, I think `store.py`'s `list` method tries loading it. I think the check for directories isn't working as intended (https://github.com/bentoml/BentoML/blob/4308c67f327d9524025e5c348ded22553824c1d4/bentoml/_internal/store.py#L69), but I'm not sure. Additionally, this happens when any file is created in the models folder, but not when folders are created.

**To Reproduce**

Steps to reproduce the issue:
1. Go to `~/bentoml/models/`
2. Create a `.DS_Store` file
3. Run `bentoml models list`

**Expected behavior**

All files should be skipped when scanning for models.

**Screenshots/Logs**

```shell
[19:12:41] WARNING  [boot] converting .DS_Store to lowercase: .ds_store         
Traceback (most recent call last):
  File "/Library/Developer/CommandLineTools/Library/Frameworks/Python3.framework/Versions/3.8/lib/python3.8/runpy.py", line 194, in _run_module_as_main
    return _run_code(code, main_globals, None,
  File "/Library/Developer/CommandLineTools/Library/Frameworks/Python3.framework/Versions/3.8/lib/python3.8/runpy.py", line 87, in _run_code
    exec(code, run_globals)
  File "/Users/spence/Library/Python/3.8/lib/python/site-packages/bentoml/__main__.py", line 4, in <module>
    create_bentoml_cli()()
  File "/Users/spence/Library/Python/3.8/lib/python/site-packages/click/core.py", line 1128, in __call__
    return self.main(*args, **kwargs)
  File "/Users/spence/Library/Python/3.8/lib/python/site-packages/click/core.py", line 1053, in main
    rv = self.invoke(ctx)
  File "/Users/spence/Library/Python/3.8/lib/python/site-packages/click/core.py", line 1659, in invoke
    return _process_result(sub_ctx.command.invoke(sub_ctx))
  File "/Users/spence/Library/Python/3.8/lib/python/site-packages/click/core.py", line 1659, in invoke
    return _process_result(sub_ctx.command.invoke(sub_ctx))
  File "/Users/spence/Library/Python/3.8/lib/python/site-packages/click/core.py", line 1395, in invoke
    return ctx.invoke(self.callback, **ctx.params)
  File "/Users/spence/Library/Python/3.8/lib/python/site-packages/click/core.py", line 754, in invoke
    return __callback(*args, **kwargs)
  File "/Users/spence/Library/Python/3.8/lib/python/site-packages/bentoml/_internal/cli/model_management.py", line 90, in list_models
    models = model_store.list(model_name)
  File "/Users/spence/Library/Python/3.8/lib/python/site-packages/bentoml/_internal/store.py", line 63, in list
    return [
  File "/Users/spence/Library/Python/3.8/lib/python/site-packages/bentoml/_internal/store.py", line 64, in <listcomp>
    ver for _d in sorted(self._fs.listdir("/")) for ver in self.list(_d)
  File "/Users/spence/Library/Python/3.8/lib/python/site-packages/bentoml/_internal/store.py", line 67, in list
    _tag = Tag.from_taglike(tag)
  File "/Users/spence/Library/Python/3.8/lib/python/site-packages/bentoml/_internal/types.py", line 221, in from_taglike
    return cls.from_str(taglike)
  File "/Users/spence/Library/Python/3.8/lib/python/site-packages/bentoml/_internal/types.py", line 226, in from_str
    return cls(tag_str, None)
  File "/Users/spence/Library/Python/3.8/lib/python/site-packages/bentoml/_internal/types.py", line 180, in __init__
    validate_tag_str(lname)
  File "/Users/spence/Library/Python/3.8/lib/python/site-packages/bentoml/_internal/utils/validation.py", line 29, in validate_tag_str
    raise InvalidArgument(f"{value} is not a valid tag: " + ", and ".join(errors))
bentoml.exceptions.InvalidArgument: .ds_store is not a valid tag: a tag's name or version must consist of alphanumeric characters, '_', '-', or '.', and must start and end with an alphanumeric character
```

**Environment:**
 - OS: MacOS 12.2.1
 - Python 3.8.9
 - BentoML 1.0.0a4

**Additional context**

MacOS creates .DS_Store files automatically, so I ran into this bug after opening the folder and sorting the files 😅

### PR DESCRIPTION
**Pull request by sauyon**
Fixes #2297 

This was mostly an oversight on my part.

### CODE REVIEW

File: tests/unit/_internal/test_store.py [hunk 1]
```diff
@@ -66,6 +66,8 @@ def __init__(self, base_path: "t.Union[PathType, FS]"):
 def test_store(tmpdir: "Path"):
     store = DummyStore(tmpdir)
 
+    os.open(os.path.join(tmpdir, ".DS_store")).close()
```
Comment (aarnphm) [inline tests/unit/_internal/test_store.py:69]: need to import `os` here

Comment (sauyon) [reply tests/unit/_internal/test_store.py:69]: It doesn't even specify `'a'` lol.

Comment (aarnphm) [reply tests/unit/_internal/test_store.py:69]: pep8 bots not me :)

**Comment by parano** [comment]
This reminds me that our documentation should really ask users not to modify or delete any filesunder `~/bentoml` directory manually 😓 one small change might break the `bentoml list` command entirely

**Review by aarnphm — APPROVED** [review]

**Comment by aarnphm** [comment]
> This reminds me that our documentation should really ask users not to modify or delete any filesunder `~/bentoml` directory manually 😓 one small change might break the `bentoml list` command entirely

Should we make `~bentoml` permission only read? users shouldn't be able to have permission to write anything under `~bentoml`, that way it would fix these problems? 

If users want to write or do anything we may ask them to copy elsewhere? For most usecase I don't think user should have permission to write at all

**Comment by parano** [comment]
@sauyon you may need to upgrade your local `black`'s version to the recent stable release

**Comment by parano** [comment]
> @sauyon you may need to upgrade your local `black`'s version to the recent stable release

Actually I think that's an issue with main branch, will fix that separately, merging this PR now!

**Pull request merged**